### ISSUE TITLE
Reasons behind padding value = 1.0

### ISSUE THREAD

**Issue by phizaz** [pytorch/text#320]
It seems to me that the default padding value for Pytorch is 0.0 as apparent here:

```
torch.nn.utils.rnn.pad_packed_sequence(sequence, batch_first=False, padding_value=0.0, total_length=None)
```

But, the padding value of Pytorch's text is rather 1.0:

https://github.com/pytorch/text/blob/7f0ed40229d680e64dc3fefbb39f2dccc255eb07/torchtext/data/field.py#L254

`self.pad_token` comes at the second place resulting in second in `itos` and hence padding value of 1.

I think having two different padding values is a bad idea? Even though I presume they won't really mix up together.

**Comment by mttk** [comment]
These are two different things -- `padding_value` in `pad_packed_sequence` is just the value which is used to **fill** the pad vectors, while the `pad_token` in `itos` is simply the **index** of the pad token in the vocabulary. The vector for the pad token is AFAIK initialized to zeros.

Usually, you pass `padding_idx=1` to `nn.Embedding` when using torchtext so the vector at that index is not trained.

**Comment by phizaz** [comment]
Yes, it is the index in the `itos`. But, is it not the same as the pad value? 

For example, I loaded a batch of IMDB dataset here, and I got a tensor of shape: (544, 32). That means the longest sequence in this batch has the length of 544. 

I printed the first sequence in the batch, I saw something like this: [value, value, ...., 1, 1, 1]. This suggests that the padded value is indeed 1. 

I agree that what ever values padded here is not relevant, I can get the training right anyway.

**Comment by mttk** [comment]
Yeah, each sentence is padded into a fixed size, but the numbers in that `(544, 32)` tensor are vocabulary (`itos`) indices, not the actual vectors used in training. 

You then usually convert them to actual word vectors by passing that tensor through a `nn.Embedding`, which results in a `(544, 32, d)` tensor, where the first dimension is the sequence length, the second the batch size and last the vector dimension. Here is where the padding index is converted to a d-dimensional zero vector.

**Comment by phizaz** [comment]
Oh, that's embarrassing! I got that now. Sorry for your inconvenience.

**Comment by phizaz** [comment]
I found it! the `<pad>` token will be considered as `<unk>` token in `Vectors` and thus has the default value of zero tensor. That's by no means straightforward for me.
